modules/aws/cloudfront-s3/variables.tf

```diff
@@ -0,0 +1,171 @@
+variable "default_tags" {
+  description = "Default Tags"
+  type        = map(string)
+}
+
+variable "aws_region" {
+  description = "aws region"
+  type        = string
+}
+
+variable "app_name" {
+  description = "argo app name"
+  type        = string
+}
+
+/* S3 Origin Parameters */
+
+variable "create_origin_access_identity" {
+  description = "Controls if CloudFront origin access identity should be created"
+  type        = bool
+  default     = true
+}
+
+variable "comment" {
+  description = "Any comments you want to include about the distribution."
+  type        = string
+  default     = "Cloudfront deployed by argonaut dev team"
+}
+
+/* CF Origin Bucket */
+
+variable "cf_origin_create_bucket" {
+  description = "Whether or not create cf origin bucket. If say 'true' bucket would be created with the name what ever you give in 'cf_origin_bucket_name'"
+  type        = bool
+}
+
+variable "cf_origin_bucket_name" {
+  description = "Name for cf origin bucket to while creating or Pass an existing bucket name when 'cf_origin_create_bucket' is set to false"
+  type        = string
+}
+
+variable "attach_policy" {
```

Comment:
@suryaoruganti 
It has good description

Reply:
1. If it is set to false, does the policy not get attached? That is what I infer from the variable name and description.
2. The expected behavior is that we automatically process this logic without user input. If the policy exists, append to it, else create one.

Reply:
Error: 

│ Error: failed getting S3 bucket policy (cf-accesslog-storage-eminent-skunk): couldn't find resource
│ 
│   with data.aws_s3_bucket_policy.existing_bucket_policy,
│   on main.tf line 92, in data "aws_s3_bucket_policy" "existing_bucket_policy":
│   92: data "aws_s3_bucket_policy" "existing_bucket_policy" {
│ 
╵
Data resource failing to get information of existing bucket if no policy is present

Reply:
so it should always be set to `attach policy`